terraform/publisher/main.tf

```diff
@@ -0,0 +1,350 @@
+  to_port   = var.container_ingress_port
+  protocol  = "tcp"
+
+  # Which security group is the rule assigned to
+  security_group_id = aws_security_group.public_service.id
+
+  # Which security group can use this rule
+  source_security_group_id = aws_security_group.public_alb.id
+}
+
+resource "aws_security_group" "public_alb" {
+  name        = "fargate_${var.service_name}_public_elb"
+  vpc_id      = data.aws_vpc.vpc.id
+  description = "Public ALB ingress and egress security group for ${var.service_name} ECS service"
+
+  ingress {
+    description = "${var.service_name} can be spoken to by the Internet"
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_security_group" "public_service" {
+  name        = "fargate_public_${var.service_name}_elb_access"
+  vpc_id      = data.aws_vpc.vpc.id
+  description = "Access to the fargate ${var.service_name} service from its public ELB"
+}
+
+#
+# Redis
+#
+
+resource "aws_security_group" "publisher_dependencies" {
+  name        = "fargate_${var.service_name}_app"
+  vpc_id      = data.aws_vpc.vpc.id
+  description = "${var.service_name} service dependencies"
+}
+
+resource "aws_security_group_rule" "ingress_redis" {
+  type      = "ingress"
+  from_port = 6379
+  to_port   = 6379
+  protocol  = "tcp"
+
+  # Which security group is the rule assigned to
+  security_group_id = var.redis_security_group_id
+
+  # Which security group can use this rule
+  source_security_group_id = aws_security_group.publisher_dependencies.id
+}
+
+#
+# DocumentDB
+#
+resource "aws_security_group_rule" "ingress_documentdb" {
+  type      = "ingress"
+  from_port = 27017
+  to_port   = 27017
+  protocol  = "tcp"
+
+  # Which security group is the rule assigned to
+  security_group_id = var.documentdb_security_group_id
+
+  # Which security group can use this rule
+  source_security_group_id = aws_security_group.publisher_dependencies.id
+}
+
+#
+# DNS
+#
+
+data "aws_route53_zone" "internal" {
+  name         = var.internal_domain_name
+  private_zone = true
+}
+
+data "aws_route53_zone" "public" {
+  name         = var.public_domain_name
+  private_zone = false
+}
+
+resource "aws_route53_record" "internal_service_names" {
+  zone_id = data.aws_route53_zone.internal.zone_id
+  name    = "${var.service_name}.${var.internal_domain_name}"
+  type    = "CNAME"
+  records = ["${var.service_name}.pink.${var.internal_domain_name}"]
+  ttl     = "300"
+}
+
+
+resource "aws_route53_record" "internal_service_record" {
+  zone_id = data.aws_route53_zone.internal.zone_id
+  name    = "${var.service_name}.pink.${var.internal_domain_name}"
+  type    = "A"
+
+  alias {
+    name                   = aws_lb.internal_alb.dns_name
+    zone_id                = aws_lb.internal_alb.zone_id
+    evaluate_target_health = false
```

Comment:
There's no reason. Initially it was for consistency, but also because we thought this might be the reason that DNS lookups were giving inconsistent results, until we realised we had to update DNS records in both AWS *and* GCP.